geoips/commandline/geoips_command.py

```diff
@@ -303,59 +331,192 @@ def _get_registry_by_interface_and_package(self, interface, package_name):
-                    interface_registry[plugin_key]["package"],
-                    interface.name,
-                    interface.interface_type,
-                    interface_registry[plugin_key]["family"],
-                    plugin_key,
-                    "N/A",
-                    interface_registry[plugin_key]["relpath"],
-                ]
-                table_data.append(plugin_entry)
-        headers = [
-            "GeoIPS Package",
-            "Interface",
-            "Interface Type",
-            "Family",
-            "Plugin Name",
-            "Source Names",
-            "Relative Path",
-        ]
+        default_headers = {
+            "package": "GeoIPS Package",
+            "interface": "Interface Name",
+            "plugin_type": "Interface Type",
+            "family": "Family",
+            "plugin_name": "Plugin Name",
+            "source_names": "Source Names",
+            "relpath": "Relative Path",
+        }
+        headers = self._get_headers_by_command(args, default_headers)
+        table_data = self._generate_table_data_by_interface(
+            interface,
+            interface_registry,
+            headers,
+        )
         print(
             tabulate(
                 table_data,
-                headers=headers,
+                headers=headers.values(),
                 tablefmt="rounded_grid",
                 maxcolwidths=self.terminal_width // len(headers),
             )
         )
+
+    def _generate_table_data_by_interface(self, interface, interface_registry, headers):
+        """Generate the table data needed for output in '_print_plugins_short_format.
+
+        Given a certain interface and a list of headers, generate a list of table data
+        used for outputting information used by 'geoips list interface' and
+        'geoips list plugins'.
+
+        Parameters
+        ----------
+        interface: geoips.interfaces.<interface_type>
+            - The interface object we will be gathering data from.
+        interface_registry: dict
+            - The plugin registry associated with the given interface.
+        headers: dict
+            - Dictionary of strings representing the key-value mapping of the
+              headers we'd like to output.
+
+        Returns
+        -------
+        table_data: 2D List of Strings
+            - The corresponding data we will output, where each sub-list is ordered
+              by headers.
+        """
+        table_data = []
+        if interface.name == "products":
+            for plugin_key in sorted(interface_registry.keys()):
+                product_dict = interface_registry[plugin_key]
+                table_data += self._get_entry(product_dict, headers)
+        else:
+            table_data += self._get_entry(interface_registry, headers)
+        return table_data
+
+    def _get_entry(self, plugin_dict, headers):
+        """Retrieve the appropriate plugin entry given a list of valid headers.
```

Comment:
Is this "retrieving"? It seems like it is "getting" or "appending".

It's not really clear from the docstring what this does or how/where it is used.

Reply:
I can update this docstring.

Reply:
Great, thanks!